<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>621</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>600</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QFormLayout" name="formLayout">
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Выберите алгоритм:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QComboBox" name="methodSelect">
        <item>
         <property name="text">
          <string>ЦДА</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Брезенхем целочисленный</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Брезенхем вещественный</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Брезенхем с устранением ступенчатости</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Ву</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Библиотечный</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:700;&quot;&gt;Отрезок:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="textFormat">
         <enum>Qt::AutoText</enum>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLabel" name="yLabel1">
        <property name="minimumSize">
         <size>
          <width>20</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Y&lt;span style=&quot; vertical-align:sub;&quot;&gt;н &lt;/span&gt;:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QDoubleSpinBox" name="xBegin">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="minimum">
         <double>-2000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.990000000000009</double>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QDoubleSpinBox" name="yBegin">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="minimum">
         <double>-2000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.990000000000009</double>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="xLabel2">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;X&lt;span style=&quot; vertical-align:sub;&quot;&gt;К &lt;/span&gt;:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLabel" name="yLabel2">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Y&lt;span style=&quot; vertical-align:sub;&quot;&gt;К &lt;/span&gt;:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QDoubleSpinBox" name="xEnd">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="minimum">
         <double>-2000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.990000000000009</double>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QDoubleSpinBox" name="yEnd">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="minimum">
         <double>-2000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.990000000000009</double>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <widget class="QPushButton" name="drawLineButton">
        <property name="text">
         <string>Построить отрезок</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:700;&quot;&gt;Спектр:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="xLabel3">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;X&lt;span style=&quot; vertical-align:sub;&quot;&gt;Ц&lt;/span&gt; :&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="11" column="1">
       <widget class="QLabel" name="yLabel3">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Y&lt;span style=&quot; vertical-align:sub;&quot;&gt;Ц&lt;/span&gt; :&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QDoubleSpinBox" name="spectrumX">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="minimum">
         <double>-2000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.990000000000009</double>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QDoubleSpinBox" name="spectrumY">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="minimum">
         <double>-2000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.990000000000009</double>
        </property>
       </widget>
      </item>
      <item row="13" column="0">
       <widget class="QLabel" name="rLabel">
        <property name="maximumSize">
         <size>
          <width>50</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Радиус:</string>
        </property>
       </widget>
      </item>
      <item row="13" column="1">
       <widget class="QLabel" name="angleLabel">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Угол в градусах:</string>
        </property>
       </widget>
      </item>
      <item row="14" column="0">
       <widget class="QDoubleSpinBox" name="spectrumR">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.990000000000009</double>
        </property>
       </widget>
      </item>
      <item row="14" column="1">
       <widget class="QDoubleSpinBox" name="spectrumStep">
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>100</height>
         </size>
        </property>
        <property name="minimum">
         <double>-2000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>2000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="15" column="0" colspan="2">
       <widget class="QPushButton" name="drawSpectrumButton">
        <property name="text">
         <string>Построить спектр</string>
        </property>
       </widget>
      </item>
      <item row="16" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="17" column="0">
       <widget class="QPushButton" name="lineColorButton">
        <property name="text">
         <string>Выберите цвет рисования:</string>
        </property>
       </widget>
      </item>
      <item row="17" column="1">
       <widget class="QLabel" name="lineColorLabel">
        <property name="maximumSize">
         <size>
          <width>21</width>
          <height>21</height>
         </size>
        </property>
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="18" column="0">
       <widget class="QPushButton" name="bgColorButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Выберите цвет фона:</string>
        </property>
       </widget>
      </item>
      <item row="18" column="1">
       <widget class="QLabel" name="bgColorLabel">
        <property name="maximumSize">
         <size>
          <width>21</width>
          <height>21</height>
         </size>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="19" column="0">
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="20" column="0" colspan="2">
       <widget class="QPushButton" name="compare_stepsButton">
        <property name="text">
         <string>Сравнение по количеству ступенек</string>
        </property>
       </widget>
      </item>
      <item row="21" column="0" colspan="2">
       <widget class="QPushButton" name="compare_timeButton">
        <property name="text">
         <string>Сравнение по времени</string>
        </property>
       </widget>
      </item>
      <item row="22" column="0">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="23" column="0" colspan="2">
       <widget class="QPushButton" name="undoButton">
        <property name="text">
         <string>Шаг назад</string>
        </property>
       </widget>
      </item>
      <item row="24" column="0" colspan="2">
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>Очистить экран</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="xLabel1">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>60</width>
          <height>16777208</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;X&lt;span style=&quot; vertical-align:sub;&quot;&gt;Н &lt;/span&gt;:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGraphicsView" name="graphicsView">
        <property name="minimumSize">
         <size>
          <width>609</width>
          <height>549</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1002</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Меню</string>
    </property>
    <addaction name="progInfo"/>
    <addaction name="authorInfo"/>
    <addaction name="separator"/>
    <addaction name="exit"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <action name="progInfo">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
  <action name="authorInfo">
   <property name="text">
    <string>Об авторе</string>
   </property>
  </action>
  <action name="exit">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
